### PR TITLE
[CP-3075] Implemented entities store

### PR DESCRIPTION
**Pull request by mkurczewski**
JIRA Reference: [CP-3075]

### :memo: Description ️

- ENTITIES_METADATA endpoint
- ENTITIES_DATA endpoint

### :muscle: Checklist before requesting a review - nice to have

- getState function in async thunk actions is correctly typed
- redux selectors are used in components / prop drilling is reduce

### :exclamation: Checklist before merging a pull request

- change went through the QA process
- translations are updated in dedicated application
- [CHANGELOG.md](./CHANGELOG.md) is updated

[CP-3075]: https://appnroll.atlassian.net/browse/CP-3075?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by dkarski — DISMISSED** [review]

File: libs/device/models/src/lib/entities/entities-data-get.validator.test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+} from "./entities-data-get.validator"
+
+describe("entitiesJsonDataValidator", () => {
+  it("validates a correct JSON data structure", () => {
+    const validJsonData = { data: [{ id: "1", name: "Entity1", 3: "value" }] }
+    expect(() => entitiesJsonDataValidator.parse(validJsonData)).not.toThrow()
+  })
+
+  it("validates a correct JSON data with empty data array", () => {
+    const invalidJsonData = { data: [] }
```
[the remark this answers is not in the document]
Comment (mkurczewski) [reply libs/device/models/src/lib/entities/entities-data-get.validator.test.ts]: Right, fixed in [07471a2](https://github.com/mudita/mudita-center/pull/2049/commits/07471a2a1c9205ac50366190d685db9f3b1e06bc)

**Review by dkarski — DISMISSED** [review]

File: libs/generic-view/store/src/lib/entities/reducer.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  builder.addCase(setEntitiesData, (state, action) => {
+    state.entities[action.payload.entityType] = {
+      ...state.entities[action.payload.entityType],
+      data: action.payload.data,
+    }
+  })
+  builder.addCase(setEntityData, (state, action) => {
+    state.entities[action.payload.entityType]?.data?.map((entity) => {
+      if (
+        entity[state.entities.idFieldKey as keyof typeof entity] ===
```
Comment (dkarski) [inline libs/generic-view/store/src/lib/entities/reducer.ts:51]: Shouldn't idFieldKey be accessed via state.entities[action.payload.entityType]?.idFieldKey?

```suggestion
        state.entities[action.payload.entityType]?.idFieldKey] ===
```

File: libs/generic-view/store/src/lib/entities/reducer.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    }
+  })
+  builder.addCase(setEntitiesData, (state, action) => {
+    state.entities[action.payload.entityType] = {
+      ...state.entities[action.payload.entityType],
+      data: action.payload.data,
+    }
+  })
+  builder.addCase(setEntityData, (state, action) => {
+    state.entities[action.payload.entityType]?.data?.map((entity) => {
```
Comment (dkarski) [inline libs/generic-view/store/src/lib/entities/reducer.ts:49]: Shouldn't the result of the map function be assigned back to state.entities[action.payload.entityType]?.data?

File: libs/device/feature/src/lib/api-entities/api-entities.service.ts [hunk 4]
```diff
@@ -43,34 +55,149 @@ export class APIEntitiesService {
...
+  @IpcEvent(APIEntitiesServiceEvents.EntitiesDataGet)
+  public async getEntitiesData({
+    entityType,
+    entityId,
+    responseType,
+    deviceId,
+  }: {
+    entityType: string
+    responseType: "json" | "file"
+    entityId?: string | number
```
Comment (dkarski) [inline libs/device/feature/src/lib/api-entities/api-entities.service.ts:129]: Wouldn't it be better to only use a string here? I see that in other places and in the documentation we consistently operate with strings.

Comment (mkurczewski) [reply libs/device/feature/src/lib/api-entities/api-entities.service.ts:129]: We are, but it's generic and numeric IDs are also a valid way of indexing items.

File: libs/generic-view/store/src/lib/entities/reducer.ts (hunk 2 above)
Comment (mkurczewski) [reply libs/generic-view/store/src/lib/entities/reducer.ts]: It should 😉 Fixed in [1c1a314](https://github.com/mudita/mudita-center/pull/2049/commits/1c1a314cf61b453ead2db9a88e1e811af84cb23c)

File: libs/generic-view/store/src/lib/entities/reducer.ts (hunk 3 above)
Comment (mkurczewski) [reply libs/generic-view/store/src/lib/entities/reducer.ts]: Right, I refactored it a little [1c1a314](https://github.com/mudita/mudita-center/pull/2049/commits/1c1a314cf61b453ead2db9a88e1e811af84cb23c)

**Review by dkarski — APPROVED** [review]

File: libs/generic-view/store/src/lib/entities/reducer.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  config?: EntitiesConfig
+  data?: EntityData[]
+  metadata?: EntitiesMetadata
+}
+
+interface EntitiesState {
+  entities: {
+    [key: EntitiesType]: Entities | undefined
+  }
+}
```
Comment (mkurczewski) [inline libs/generic-view/store/src/lib/entities/reducer.ts:28]: Refactor to
```TS
interface EntitiesState {
 [deviceId]: {
    [key: EntitiesType]: Entities | undefined
  }
}
```

File: libs/generic-view/store/src/lib/entities/reducer.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  setEntitiesData,
+  setEntitiesMetadata,
+  setEntityData,
+} from "./actions"
+
+type EntitiesType = string
+
+interface Entities {
+  idFieldKey?: string
+  config?: EntitiesConfig
```
Comment (mkurczewski) [inline libs/generic-view/store/src/lib/entities/reducer.ts:19]: Remove `?`

**Review by dkarski — DISMISSED** [review]

**Pull request merged**